src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -630,46 +635,23 @@ export const SwapProvider = ({ children }: SwapProviderProps) => {
       return { icon: '􀕹', label: review, disabled: true };
     }
 
-    const inputAsset = internalSelectedInputAsset.value;
-    const sellAmount = (() => {
-      const inputAmount = SwapInputController.inputValues.value.inputAmount;
-      if (!quote.value || 'error' in quote.value) return inputAmount;
-      return quote.value.sellAmount.toString();
-    })();
-
+    const sellAsset = internalSelectedInputAsset.value;
     const enoughFundsForSwap =
-      inputAsset && lessThanOrEqualToWorklet(sellAmount, toScaledIntegerWorklet(inputAsset.balance.amount, inputAsset.decimals));
+      sellAsset && lessThanOrEqualToWorklet(SwapInputController.inputValues.value.inputAmount, sellAsset.maxSwappableAmount);
```

Comment:
NOTE: this is currently INCORRECT and WIP
issue: inputAmount is the "nicely formatted" version of maxSwappableAmount currently, so it will be rounded up and therefore incorrect in this scenario